rclcpp/src/rclcpp/executors/multi_threaded_executor.cpp

```diff
@@ -42,15 +42,17 @@ MultiThreadedExecutor::spin()
   }
   RCLCPP_SCOPE_EXIT(this->spinning.store(false); );
   std::vector<std::thread> threads;
-  {
+  size_t thread_id = 1;
+  if (number_of_threads_ > 0) {
```

Comment:
According to the constructor `number_of_threads_` can't be zero. Even if the condition is checked here there will be one exector (namely this thread). Therefore I would suggest to remove the condition and if you want to be defensive add a runtime assertion that `number_of_threads_` > 0.